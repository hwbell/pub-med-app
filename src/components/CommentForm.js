import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button, Input, Form, InputGroup, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

// this form is toggled visible by the 'add to collection' button on the search page
class CommentForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      comment: {},
    }

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  componentDidMount() {
    // 
  }

  // handleChange changes the value of the input
  handleChange(e) {
    let comment = this.state.comment;

    comment.text = e.target.value;
    this.setState({ comment }, () => {
      console.log(this.state.comment)
    })
  }

  // handleSubmit will fire the email
  handleSubmit() {

    if (this.state.comment.text) {
      this.props.confirm(this.state.comment);
    }
    this.props.toggle();
  }

  render() {

    return (
      <div style={{ marginTop: '25px' }}>

        <p style={styles.text}>
          Have something to say about this thread? Add a comment to join the discussion.
        </p>


        <Input style={styles.input}
          type="textarea"
          placeholder={"Your comment here"}
          value={this.state.name}
          onChange={(e) => this.handleChange(e)}
        />

        <div style={styles.buttonHolder}>
          <Button className="add article-button" size="sm"
            onClick={this.handleSubmit}>post comment</Button>
          <Button className="back article-button" 
            onClick={this.props.toggle} size="sm">cancel</Button>
        </div>

      </div>

    );
  }
}

const styles = {
  content: {
    height: '100%',
  },
  text: {
    color: 'black',
    margin: '5px 20px'
  },
  input: {
    width: '95%',
    margin: '15px auto',
    borderRadius: '6px',
    padding: '8px'
  },
  buttonHolder: {
    alignSelf: 'flex-end',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  }

}

export default CommentForm;
