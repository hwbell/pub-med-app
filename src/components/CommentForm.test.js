import React from 'react';

import CommentForm from './CommentForm';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

const confirmStub = jest.fn();
const toggleStub = jest.fn();

const someProps = {
  confirm: confirmStub,
  toggle: toggleStub
};

describe('CommentForm', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CommentForm {...someProps} />)
  })

  it('renders without crashing', () => {
    wrapper.update();
  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<CommentForm  {...someProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains the correct elements', () => {
    expect(wrapper.find('Input').length).toBe(1);
    expect(wrapper.find('p').length).toBe(1);
    expect(wrapper.find('.add').length).toBe(1);
    expect(wrapper.find('.back').length).toBe(1);
  })

  it('should change the value of this.state.comment when handleChange() is called', () => {
    let e = {
      target: {
        value: 'This is a comment!'
      }
    }

    wrapper.instance().handleChange(e);
    wrapper.update();
    expect(wrapper.state().comment.text).toBe(e.target.value)
  })

  it('should fire this.handleSubmit() and this.props.confirm() when post thread button is clicked', () => {
    jest.clearAllMocks();

    wrapper.find('.add').simulate('click');
    wrapper.update();


    expect(toggleStub.mock.calls.length).toBe(1);

    // without a comment it shouldn't be called
    expect(confirmStub.mock.calls.length).toBe(0);

    // provide a comment 
    wrapper.setState({
      comment: { text: 'Yo dude!' }
    });
    wrapper.find('.add').simulate('click');
    wrapper.update();
    expect(confirmStub.mock.calls.length).toBe(1);

  })

  it('should fire this.props.toggle() when the cancel button is clicked', () => {

  })
})