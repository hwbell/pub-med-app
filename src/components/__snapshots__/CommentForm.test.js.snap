// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm renders correctly 1`] = `
<div
  style={
    Object {
      "marginTop": "25px",
    }
  }
>
  <p
    style={
      Object {
        "color": "black",
        "margin": "5px 20px",
      }
    }
  >
    Have something to say about this thread? Add a comment to join the discussion.
  </p>
  <textarea
    className="form-control"
    onChange={[Function]}
    placeholder="Your comment here"
    style={
      Object {
        "borderRadius": "6px",
        "margin": "15px auto",
        "padding": "8px",
        "width": "95%",
      }
    }
  />
  <div
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <button
      aria-label={null}
      className="add article-button btn btn-secondary btn-sm"
      onClick={[Function]}
      type="button"
    >
      post comment
    </button>
    <button
      aria-label={null}
      className="back article-button btn btn-secondary btn-sm"
      onClick={[Function]}
      type="button"
    >
      cancel
    </button>
  </div>
</div>
`;
